client/src/Routes.js

```diff
@@ -5,16 +5,20 @@ import Home from "./pages/Home";
 import CreateProfile from "./pages/CreateProfile";
 import ViewProfile from "./pages/ViewProfile";
 import EditProfile from "./pages/EditProfile";
+import NotEligible from './pages/NotEligible'
+
+import { ProtectedRoute } from './authentication/ProtectedRoute'
 
 const Routes = () => {
 	return (
 		<BrowserRouter>
 			<Switch>
 				<Route exact path='/' component={Home} />
+				<Route path='/viewdetail' component={ViewProfile} />
 				<Route path='/createprofile' component={CreateProfile} />
 				<Route path='/viewprofile' component={ViewProfile} />
 				<Route path='/editprofile' component={EditProfile} />
```

Comment:
What's the difference between `/viewprofile` and `/viewdetail`?
Also, I would change the other two:
- instead of `/newprofile` I'd use `profiles/new`
- instead of `/editprofile` I'd use `profiles/:id/edit` - this way you now from the url param which user you're editing